rust/src/context.rs

```diff
@@ -529,3 +528,105 @@ pub fn add_to_linker<T: 'static>(
     linker.allow_unknown_exports(true);
     Ok((events_tx, events_rx))
 }
+
+#[cfg(feature = "component-model")]
+pub mod component {
+    use super::*;
+    use wasmtime::component::Linker;
+
+    wasmtime::component::bindgen!({
+        path: "../wit",
+        async: false
+    });
+
+    use dylibso::observe::api::{ MetricFormat, LogLevel, Host as ApiHost };
+    use dylibso::observe::instrument::Host as InstrumentHost;
+
+    pub struct ObserveSdkBindings {
+        pub(crate) instr_context: Arc<Mutex<InstrumentationContext>>,
+        pub(crate) wasm_instr_info: WasmInstrInfo
+    }
+
+    impl TryInto<super::MetricFormat> for MetricFormat {
```

Comment:
`wasmtime::component::bindgen` is doing some heavy lifting for us – giving us Real Enums and Strings™, etc – but that means we have to map these generated types into our internal domain types.